tests/bluetooth/audio/mocks/include/expects_util.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2023 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef MOCKS_UTIL_H_
+#define MOCKS_UTIL_H_
+
+#include <zephyr/types.h>
+#include <zephyr/sys/util_macro.h>
+
+#define CHECK_EMPTY(_x) UTIL_BOOL(IS_EMPTY(_x))
+#define COND_CODE_EMPTY(_x, _if_any_code, _else_code)                                              \
+	COND_CODE_1(CHECK_EMPTY(_x), _if_any_code, _else_code)
+#define IF_EMPTY(_a, _code) COND_CODE_EMPTY(_a, _code, ())
+#define IF_NOT_EMPTY(_a, _code) COND_CODE_EMPTY(_a, (), _code)
+
+#define expect_data(_func_name, _arg_name, _exp_data, _data, _len)                                 \
+	IF_NOT_EMPTY(_exp_data, (expect_data_equal(_func_name, _arg_name, _exp_data, _data, _len);))
+
+static inline void expect_data_equal(const char *func_name, const char *arg_name,
+				     const uint8_t *expect, const uint8_t *data, size_t len)
+{
+	for (size_t i = 0U; i < len; i++) {
+		zexpect_equal(expect[i], data[i],
```

Comment:
Did you consider `zexpect_mem_equal`?

Reply:
Yes, I used it, but in this particular case iteration gives more information about the PDU entries that are different than expected. It's easier to catch the problem looking into the logs

Reply:
That makes sense. That could be added as a comment in the code :)

Have you considered casting the data to a struct and have a struct compare instead? That may be even better for debugging purposes

Reply:
I think it's sufficient as it is. The expected and test data are just uint8_t buffers so it's quite easy to figure out the data that failed based on array index.